TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -20,6 +20,21 @@ public class ProgramClassBuilder : IProgramClassBuilder
     {
         private static bool _SourceCodeDumped;
 
+        internal void LogAnomalousLineIndex()
+        {
+            const string message = "Anomalous line index found !";
+            if (_SourceCodeDumped)
+            {
+                // Full source with changes has already been dumped, just add a simple trace to help notice the line index problem
+                LoggingSystem.LogMessage(LogLevel.Error, message);
```

Comment:
I don't think we need a log once the full source has been dumped.
If there is a problem with LineIndex, it will remain till the source is fully parsed again. We could then end up with a lot of logs.

Reply:
You're right but what about adding a second flag to add this log just once ?

Reply:
I've kept the trace because the invalid AST problems are detected _during_ the AST building and the line index problem are reported _after_ the AST building.
If we don't add a specific trace for wrong line indices we'll likely only have traces for 'invalid parent for paragraph' and we may not remark the lines.